app/models/raw_column_transaction.rb

```diff
@@ -20,11 +20,18 @@ class RawColumnTransaction < ApplicationRecord
   after_create :canonize, if: -> { canonical_transaction.nil? }
 
   def canonize
-    create_canonical_transaction!(
+    ct = create_canonical_transaction!(
       amount_cents:,
       memo:,
       date: date_posted,
     )
+
+    raw_pending_column_transaction = RawPendingColumnTransaction.find_by(column_id: column_transaction["transaction_id"])
```

Comment:
- **question**: can there ever be a `column_transaction` without `transaction_id`?
  ```suggestion
      raw_pending_column_transaction = RawPendingColumnTransaction.find_by(column_id: column_transaction.fetch("transaction_id"))
  ```
- **question**: can we ever fail to find a matching record here?
  ```suggestion
      raw_pending_column_transaction = RawPendingColumnTransaction.find_by!(column_id: column_transaction["transaction_id"])
  ```

Reply:
`transaction_id` should never be null. 

Yes we can fail to find a matching record - I'll add that handling